plugins/git/functions

```diff
@@ -4,6 +4,23 @@ source "$PLUGIN_CORE_AVAILABLE_PATH/common/functions"
 source "$PLUGIN_AVAILABLE_PATH/apps/functions"
 source "$PLUGIN_AVAILABLE_PATH/config/functions"
 
+use_git_worktree() {
+    declare desc="detects whether to use git worktree"
+    local GIT_VERSION MAJOR_VERSION MINOR_VERSION
+
+    GIT_VERSION=$(git --version | awk '{split($0,a," "); print a[3]}')
+    MAJOR_VERSION=$(echo "$GIT_VERSION" | awk '{split($0,a,"."); print a[1]}')
+    MINOR_VERSION=$(echo "$GIT_VERSION" | awk '{split($0,a,"."); print a[2]}')
+
+    if [[ "$MAJOR_VERSION" -ge "3" ]]; then
+        return 0
+    elif [[ "$MAJOR_VERSION" -eq "2" ]] && [[ "$MINOR_VERSION" -ge "11" ]]; then
```

Comment:
I did not have the time to test this with git 2.5, so I concentrated on git 2.11.  This way I am confident that we don't break any previously working git versions.  And up to git 2.10 should work with the old clone approach.
I can change it to 2.5 of course, but then I would recommend some more testing...